app/api/files/uploadMiddleware.ts

```diff
@@ -27,7 +27,7 @@ const singleUpload =
   (filePath?: pathFunction, storage = defaultStorage) =>
   async (req: Request, res: Response, next: NextFunction) => {
     try {
-      await new Promise((resolve, reject) => {
+      await new Promise<void>((resolve, reject) => {
```

Comment:
This promise is not always void since it can reject an error, should it be specified in the Promise type?

Reply:
i do not think a rejection counts as the value that a function can return, its equivalent to throwing an error ? when the promises executes without errors you always have a void error, otherwise you are in a error ```catch``` flow ? makes sense ?

Reply:
👍